tools/clang/test/SemaHLSL/hlsl/workgraph/template.hlsl

```diff
@@ -0,0 +1,24 @@
+// RUN: %dxc -Tlib_6_8 -verify %s
+
+// GroupNodeInputRecords may not have zero size record on DependentType.
+// GroupNodeInputRecords may not non-struct/class DependentType. 
+
+template<typename T>
+struct Record {  }; // expected-note{{zero sized record defined here}}
+
+template <typename T> void foo(GroupNodeInputRecords<T> data) {}
+
+template <typename T> void bar() {
+  GroupNodeInputRecords<Record<T> > data; // expected-error{{record used in GroupNodeInputRecords may not have zero size}}
+  foo(data);
```

Comment:
None of the calls to foo() are causing the diagnostic to fire.  I guess it's because it already fired for the local, invalidating the `data` decl?  What if something constructs a local method with an invalid argument type based on the template type, without having called that method.  For example:

```hlsl
template <typename T>
class CFoo {
  T var;
  void foo(GroupNodeInputRecords<T> data) {}
};
// ...
  CFoo<int> c;
```

Should that result in a diagnostic triggered by the `CFoo<int> c` instantiation?

Reply:
Added.